File: frontend/src/microservices/hotel/objects/HotelPlannerResponse.ts
import type { RoomTypeSummaryResponse } from './RoomTypeSummaryResponse'

export type HotelPlannerResponse = {
  hotelId: string
  hotelName: string
  location: string
  status: string
  createdAt: string
  roomTypes: RoomTypeSummaryResponse[]
}

export const hotelPlannerResponseFromJson = (json: string): HotelPlannerResponse =>
  JSON.parse(json) as HotelPlannerResponse

export const hotelPlannerResponseToJson = (value: HotelPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/HotelPlannerResponse.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class HotelPlannerResponse(
    hotelId: String,
    hotelName: String,
    location: String,
    status: String,
    createdAt: String,
    roomTypes: List[RoomTypeSummaryResponse]
)

object HotelPlannerResponse:
  given sourceEncoder: Encoder[HotelPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[HotelPlannerResponse] = deriveDecoder
